spring-ai-core/src/main/java/org/springframework/ai/vectorstore/SimpleVectorStoreContent.java

```diff
@@ -100,22 +99,10 @@ public SimpleVectorStoreContent(String id, String text, Map<String, Object> meta
 
 		this.id = id;
 		this.text = text;
-		this.metadata = Collections.unmodifiableMap(new HashMap<>(metadata));
+		this.metadata = Map.copyOf(metadata);
 		this.embedding = Arrays.copyOf(embedding, embedding.length);
 	}
 
-	/**
-	 * Creates a new instance with an updated embedding vector.
-	 * @param embedding the new embedding vector, must not be null
-	 * @return a new instance with the updated embedding
-	 * @throws IllegalArgumentException if embedding is null or empty
-	 */
-	public SimpleVectorStoreContent withEmbedding(float[] embedding) {
```

Comment:
Public methods need to be deprecated first before getting deleted. I will re-add and deprecate this method instead.